js/gestionPresupuesto.js

```diff
@@ -1,25 +1,151 @@
 // TODO: Crear las funciones, objetos y variables indicadas en el enunciado
+//Cathaysa Navarro Benítez.
+let gastos = [];
+let idGasto = 0;
+let presupuesto = 0;
 
-// TODO: Variable global
+//actualizarPresupuesto();
+//mostrarPresupuesto();
 
+//pruebas para la función CrearGasto.
+/*let gasto1 = new CrearGasto("Gasto 1");
+let gasto2 = new CrearGasto("Gasto 2", 23.55);
+let gasto3 = new CrearGasto("Gasto 3", 23.55, "2021-10-06T13:10");
+let gasto4 = new CrearGasto("Gasto 4", 23.55, "2021-10-06T13:10", "casa");
+let gasto5 = new CrearGasto("Gasto 5", 23.55, "2021-10-06T13:10", "casa", "supermercado");
+let gasto6 = new CrearGasto("Gasto 6", 23.55, "2021-10-06T13:10", "casa", "supermercado", "comida");*/
 
-function actualizarPresupuesto() {
-    // TODO
+//console.log(gasto6.mostrarGastoCompleto());
+
+function actualizarPresupuesto(dinero) {
+  if (dinero >= 0) {
+    presupuesto = dinero;
+    return presupuesto;
+  } else {
+    console.error("La cantidad introducida no es válida");
+    return -1;
+  }
 }
 
 function mostrarPresupuesto() {
-    // TODO
+  return "Tu presupuesto actual es de " + presupuesto + " €";
+}
+//FUNCIÓN CREAR GASTO:
+function CrearGasto(descripcion, valor, fecha, ...etiquetas) {
+  this.descripcion = descripcion;
+  //comprobación de si el valor tiene un tipo correcto.
+  this.valor = typeof valor === "number" && valor >= 0 ? valor : 0;
+
+  //verificación de la fecha
+  if (fecha && !isNaN(Date.parse(fecha))) {
+    this.fecha = Date.parse(fecha);
+  } else {
+    this.fecha = Date.now();
+  }
+  //el array de etiquetas comienza vacío
+  this.etiquetas = etiquetas.length > 0 ? etiquetas : [];
```

Comment:
Aquí, en lugar de añadir el array de etiquetas como si fuera válido, puedes llamar a la función `this.anyadirEtiquetas` para comprobar que no hay duplicados.